### ISSUE TITLE
Bioperl outputs invalid Genbank/EMBL

### ISSUE THREAD

**Issue by innovate-invent** [bioperl/bioperl-live#321]
Bioperl quotes `anticodon` and `transl_except` feature qualifiers when they line wrap. This is invalid.

Copying NIH statement here for reference. Some messages were removed/edited for brevity.

> Case #: CAS-380254-C8B2Z2 
> The document here: http://www.insdc.org/files/feature_table.html does not describe how and where continuation lines should be created. There is divergence between biopython, bioperl, and other tools on how this is handled causing datasets to fail to be parsed.
> For example:
> The reference genome for NC_018080 downloaded from RefSeq fails to parse.
> Features containing:
>     /anticodon=(pos:complement(1123552..1123554),aa:Leu,
>     seq:caa)
> are failing to parse, this output is consistent with biopython.
> Bioperl wraps at column 80:
>     /anticodon="(pos:complement(1123552..1123554),aa:Leu,seq:caa
>     )"
> Because there is no official statement on how wrapping should occur I can't raise issues with the appropriate software developer.
> Any input on the matter would be appreciated. 
> Case Created:  6/6/2019 4:08 PM 
> 
> ----------------------------
> 
> Dear Nolan,
> Thanks for the additional info. It is expected that long anticodon qualifiers will wrap if they exceed the 80-character max line length. Furthermore, any of the following are legal within the spec:
>     /anticodon=(pos:complement(1123552..1123554),aa:Leu,
>     seq:caa)
>     /anticodon=(pos:complement(1123552..1123554),aa:Leu,seq:
>     caa)
>     /anticodon=(pos:complement(1123552..1123554),
>     aa:Leu,seq:caa)
> What might be happening is that the parsers you're testing may be combining the /anticodon qualifier and its continuation line, but adding an extra space, which would then invalidate the qualifier. That would be a bug in the parser code. For parsing an anticodon qualifier with a continuation line, you have to:
>     Realize that the you aren't yet at the end of the qualifier value (terminal right paren)
>     Read the next line, and strip all leading white space
>     Directly append (2) to the qualifier value
> For other qualifiers with text content, a parser generally needs to include a space when appending a continuation line, unless the row ends in a hyphen or is an unbroken string (no spaces, such as with a long chemical formula or the protein sequence in a /translation qualifier). But /anticodon may need an exception, depending on how the parser code is written.
> I hope that helps clear up the bug. If you file a bug report, please let us know the ticket for our records.
> 
> ...
> 
> As our flatfile does not have quotes, the quoting is not valid. The quotes are a bug. 
> 
> Best regards,
> -Terence Murphy, Ph.D.
> RefSeq and Gene Team Lead
> NCBI/NLM/NIH/DHHS
> 
> ----------------------------

https://github.com/biopython/biopython/issues/2112

**Comment by cjfields** [comment]
Thanks @innovate-invent I'll try having a look.

**Comment by cjfields** [comment]
Cc'ing @peterjc in case there are similarities to the Biopython bug: https://github.com/biopython/biopython/issues/2112

Interesting thing: the Bioperl code does indicate that the anticodon tag/qualifier shouldn't be quoted in output but does include an exception if the line wraps at 80 chars (`%FTQUAL_NO_QUOTE` contains a list of tags that shouldn't be quoted):

https://github.com/bioperl/bioperl-live/blob/master/lib/Bio/SeqIO/genbank.pm#L1302

So, the old logic for output is to quote with a list of exceptions, but *always* quote if the feature line wraps. This particular instance suggests the logic is flawed and there are a few exceptions.

Commenting it out does partly fix this particular bug (wrapping is still off) but breaks some tests.  It *does* seem to parse as expected though (with no added spaces) based on a quick dump of the record, so the issue seems primarily in the generation of output:

```
                                      'anticodon' => [
                                                       '(pos:complement(1123552..1123554),aa:Leu,seq:caa)'
                                                     ],
```

**Comment by cjfields** [comment]
Attaching a smaller test case version of the problem feature in Genbank format (@peterjc see attachment if needed)

[NC_018080-sample.gb.txt](https://github.com/bioperl/bioperl-live/files/3638792/NC_018080-sample.gb.txt)

Note the output with fixed logic with BioPerl is still wrong:

```
    tRNA            complement(654..740)
                     /anticodon=(pos:complement(1123552..1123554),aa:Leu,seq:ca
                     a)
                     /inference="COORDINATES: profile:tRNAscan-SE:2.0.3"
                     /locus_tag="PADK2_RS05265"
                     /note="Derived by automated computational analysis using
                     gene prediction method: tRNAscan-SE."
                     /old_locus_tag="PADK2_t29613"
                     /product="tRNA-Leu"
```

**Comment by cjfields** [comment]
This seems to be partially addressed, but the text wrap in the output is still not adequate:

```
     tRNA            complement(654..740)
                     /anticodon=(pos:complement(1123552..1123554),aa:Leu,seq:ca
                     a)
                     /inference="COORDINATES: profile:tRNAscan-SE:2.0.3"
                     /locus_tag="PADK2_RS05265"
                     /note="Derived by automated computational analysis using
                     gene prediction method: tRNAscan-SE."
                     /old_locus_tag="PADK2_t29613"
                     /product="tRNA-Leu"
```

Will look into this a bit more but the feature does seem to round-trip in BioPerl.

**Comment by cjfields** [comment]
Added a fix for text wrapping, now works but I want to see if travis passes before closing.

```
     tRNA            complement(654..740)
                     /anticodon=(pos:complement(1123552..1123554),aa:Leu,
                     seq:caa)
                     /inference="COORDINATES: profile:tRNAscan-SE:2.0.3"
                     /locus_tag="PADK2_RS05265"
                     /note="Derived by automated computational analysis using
                     gene prediction method: tRNAscan-SE."
                     /old_locus_tag="PADK2_t29613"
                     /product="tRNA-Leu"
```

**Comment by cjfields** [comment]
Appears to be fixed with 7fb6792.   @innovate-invent @peterjc or others, please reopen if this doesn't address issues.
